Bottom: a097192d67b980aea88fc8a7cfbff9ea27aac8f8
Top:    480c5f6de80da50004b546aa1038f952d0f309f9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-26 14:18:37 +0100

Refresh of polygon1-c-fix-dicer-to-give-u

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index ba9d5bf..7ebf4cd 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -2201,11 +2201,6 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   region.Y1 = MAX (0, MIN (PCB->MaxHeight, region.Y1));
   region.Y2 = MAX (0, MIN (PCB->MaxHeight, region.Y2));
 
-  printf ("EV: %i %i %i %i\n", ev->area.x, ev->area.y, ev->area.width, ev->area.height);
-  printf ("PCB: %i %i %i %i\n", region.X1, region.X2, region.Y1, region.Y2);
-  if (region.X1 == region.X2)
-    *(char *)0 = 0;
-
   glColor3f (port->bg_color.red / 65535.,
              port->bg_color.green / 65535.,
              port->bg_color.blue / 65535.);
diff --git a/src/polygon.c b/src/polygon.c
index 059a1f6..bdec6b4 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1751,29 +1751,23 @@ void
 NoHolesPolygonDicer (PolygonTypePtr p, const BoxType * clip,
                      void (*emit) (PLINE *, void *), void *user_data)
 {
-  POLYAREA *save, *ans, *cur, *next;
+  POLYAREA *main_contour, *cur, *next;
 
-  ans = save = poly_Create ();
+  main_contour = poly_Create ();
   /* copy the main poly only */
-  poly_Copy1 (save, p->Clipped);
+  poly_Copy1 (main_contour, p->Clipped);
   /* clip to the bounding box */
   if (clip)
     {
       POLYAREA *cbox = RectPoly (clip->X1, clip->X2, clip->Y1, clip->Y2);
-      if (cbox)
-        {
-          int r = poly_Boolean_free (save, cbox, &ans, PBO_ISECT);
-          save = ans;
-          if (r != err_ok)
-            save = NULL;
-        }
+      poly_Boolean_free (main_contour, cbox, &main_contour, PBO_ISECT);
     }
-  if (!save)
+  if (main_contour == NULL)
     return;
   /* Now dice it up.
    * NB: Could be more than one piece (because of the clip above)
    */
-  cur = save;
+  cur = main_contour;
   do
     {
       next = cur->f;
@@ -1781,7 +1775,7 @@ NoHolesPolygonDicer (PolygonTypePtr p, const BoxType * clip,
       r_NoHolesPolygonDicer (cur, emit, user_data);
       /* NB: The POLYAREA was freed by its use in the recursive dicer */
     }
-  while ((cur = next) != save);
+  while ((cur = next) != main_contour);
 }
 
 /* make a polygon split into multiple parts into multiple polygons */
